gen/generate_test.go

```diff
@@ -131,27 +132,37 @@ func TestGenesisJsonCreation(t *testing.T) {
 	t.Parallel()
 
 	type testCase struct {
-		name string
-		gd   GenesisData
+		name             string
+		gd               GenesisData
+		expectedOverride string
 	}
 
-	// `base` is a canonical test confirming `devnet.json` generates the intended `genesis.json`.
-	base := func() testCase {
-		jsonBytes, err := os.ReadFile("devnet.json")
+	defaultGenesisFromFile := func(filename string) GenesisData {
+		jsonBytes, err := os.ReadFile(filename)
 		require.NoError(t, err)
 
 		gd := DefaultGenesis
 		err = json.Unmarshal(jsonBytes, &gd)
 		require.NoError(t, err)
 
-		return testCase{"base", gd}
+		return gd
+	}
+
+	// `base` is a canonical test confirming `devnet.json` generates the intended `genesis.json`.
+	base := func() testCase {
+		return testCase{"base", defaultGenesisFromFile("devnet.json"), ""}
 	}
 
 	// `balance` extends `base` to confirm overriding the rewards pool balance works.
 	balance := func() testCase {
 		gd := base().gd
 		gd.RewardsPoolBalance = 0 // Expect generated balance == MinBalance
-		return testCase{"balance", gd}
+		return testCase{"balance", gd, ""}
+	}
+
+	// `testnet` confirms the generated genesis.json conforms to a previously generated _installer_ artifact.
+	testnet := func() testCase {
+		return testCase{"testnet", defaultGenesisFromFile("testnet.json"), "../installer/genesis/testnet/genesis.json"}
```

Comment:
@algorandskiy Let me know how you'd like to proceed?

* I agree relative paths can be fragile though I saw the convention applied elsewhere (e.g. https://github.com/algorand/go-algorand/blob/3701d291c8d441af163eec78195f88ea170a2e88/config/config_test.go#L228), and felt for a test (rather than source) it's _reasonable_.  
* Unless the test is moved, I don't _think_ the relative path is subject to change.  Though if you prefer, I'll look into identifying the project root dir to establish an absolute path.

Reply:
oh, os.Getwd is even worse. I run this test from the repo root and ./gen, both work, good enough for me